### PR TITLE
avoid dirty hack to prometheus client

### PR DESCRIPTION
**Pull request by andyxning**
**What this PR does / why we need it**:
This PR avoids dirty code change hack to prometheus client.

### CODE REVIEW

File: pkg/collectors/pod_test.go [hunk 1]
```diff
@@ -157,93 +157,93 @@ func TestPodCollector(t *testing.T) {
 			pods: []v1.Pod{
 				{
 					ObjectMeta: metav1.ObjectMeta{
-						Name:      "pod1",
-						Namespace: "ns1",
+						Name:      "pod3",
```
Comment (brancz) [inline pkg/collectors/pod_test.go]: The fact that we need to do this to make the tests pass sounds like there is some global/cross-test state that should not be. Can you spot where this is coming from? I would rather get rid of that global state and not do all of these changes (if that's the case).

Comment (andyxning) [reply pkg/collectors/pod_test.go]: Actually, we need only three changes to one of the pod unit test cases.

Comment (andyxning) [reply pkg/collectors/pod_test.go]: /cc @brancz

**Pull request merged**